mod/node-api/engines/echo/middleware.go

```diff
@@ -21,35 +21,57 @@
 package echo
 
 import (
-	"errors"
 	"net/http"
 
+	"github.com/berachain/beacon-kit/mod/errors"
 	"github.com/berachain/beacon-kit/mod/node-api/handlers"
 	"github.com/berachain/beacon-kit/mod/node-api/handlers/types"
 	"github.com/labstack/echo/v4"
 )
 
-func buildHandler(
-	handler handlers.Route[Context],
+// ErrorResponse is a response that is returned when an error occurs.
+type ErrorResponse struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+// errorMiddleware is a middleware that converts errors to an HTTP status code
+// and response.
+func errorMiddleware(
```

Comment:
i think this should be a response handler/middleware instead of error middleware, since it handles every response type, including status OK